### DISCUSSION TITLE
Add never stop mode

### DISCUSSION THREAD

**Discussion by segman1** [thejoabo/virtualfisher-bot#40] [Ideas]
add a setting that allows you to still fish when the cmd is closed / your pc is shut downed

**Comment by thejoabo** [comment]
Hello @segman1, this isn't possible for the current setup. In order to continue fishing when your PC is turned off, it would need, regardless, a machine to run it. In this case the best solution would be a VPS. But If I buy a dedicated VPS to allow users to allocate the resources needed to run it while they are AFK, it'd be expensive for me. Therefore there is not much I can do about this "never stop mode".

About the CMD, I could remove it (to run in the background as a windows service), but would be more complicated than I planned the bot to be.

For now, I can provide these two options:

- Start minimized as default with a "run.cmd" file:
  1. Go to the autofisher directory
  2. Create a new file and rename it to "run.cmd"
  3. Edit the file using any text editor
  4. Paste the following code: `@echo off & start /min python.exe autofishbot.py ^& exit`
  5. Save the file and run it as a executable

The bad thing about this option is that the icon would be still visible in the taskbar. In the other hand you could create a shortcut of this file and run it from your desktop and skip some annoyances to open CMD (also, as a shortcut, you could change the icon and some behaviors).

- Start the autofisher in another desktop (requires > windows 10 or 11)
  1. Press **Windows + Ctrl + D** (to create a new virtual desktop)
  2. Run the autofisher as usual
  3. Press **Windows + Tab** (to open task view) and click on your primary desktop ("Desktop 1")
  4. Alternatively, you can use **Windows + Ctrl + Left** or **Right Arrow** to Switch between virtual desktops.

This option hides the icon in the taskbar.

**Reply by segman1** [reply]
ccan you add a auto daily and maybe auto boat/rod

**Reply by thejoabo** [reply]
@segman1, I can add a "auto daily" feature on the bot startup, but keep in mind that I have no way to send this command unless you have the autofisher opened (in case you meant this feature to run independently). About this "auto boat/rod" feature, you meant auto upgrades ? Can you clarify a bit more ?

**Reply by segman1** [reply]
Like for example;
on the config file you can type in what rod you want and the bot will try to buy it after it sells all, if it fails it will just continue fishing, and about the boat i do not know how to really set that up, and the auto upgrade could be pretty good  (if it can buy special upgrades) i think the bot can buy one by one or if possible check the cheapest one but i dont think thats possible plus i dont really want the worker related upgrades so how is that gonna work?

**Reply by thejoabo** [reply]
I think I got it. But the thing about Discord's API is that slash commands (and interactions) overall seems to have much restricted rate limits than normal messages, so spamming slash commands to check inventory, shop pages, values, and + would be a recipe to failures. 
What I'm trying to say is that: everything is possible, but not completely viable to do. 

One possible solution is, on startup, the bot runs a "complex" routine of essential activities. Like, updating a cached database of things to work on (instead of relying of spamming commands every time), this things would include:

- Update profile info (current money, level, fishes ...)
- Update rods info (like already bought and rods there are available to buy ...)
- Update boats info (^^^^)
- Update upgrades info (^^^^)
- Sell all inventory (as usual)
- Buy baits (as usual)
- Buy buffs (as usual, if set on user config)
- and + 

With this set up of gathered information I could implement auto upgrades/purchases of rods, boats and upgrades. It is completely possible to do, but would demand quite a bit of rewriting and some logic changes, and obviously a ton of tests. Also, I would need to add multiple options on the config file to allow users to properly choose each and every action they want to be automated (and probably rewrite this whole "_ConfigManager_" method that I already have to something more organized).

Anyways, it's a great idea and could add a lot of room to new features (like a "Shop" menu on the bot-side, way more practical then using discord's interactions), but it'll demand some effort and time. Right now, I'm finishing my semester and in the middle of another project, so I've been quite busy, regardless, **I plan to implement this feature on the next release**.

**TLDR: I'll add it when I have the time.**

**Also, this solution would be good for you ? Any suggestions/improvements ?**

**Reply by segman1** [reply]
damn bro, thank you btw

and yes abit too much but yes

**Comment by Maxhem2** [comment]
Buy a server on strato.de and cancel it within 30 days, repeat also can be automated with 2captcha.
